pallets/portfolio/src/lib.rs

```diff
@@ -352,6 +353,16 @@ decl_module! {
                     .filter(|(identity, _, number)| !Portfolios::contains_key(identity, number))
                     .for_each(|(identity, name, _)| NameToNumber::remove(identity, name));
             });
+            storage_migrate_on!(StorageVersion::get(), 2, {
+                NameToNumber::iter()
+                    .filter(|(identity, _, number)| !Portfolios::contains_key(identity, number))
```

Comment:
The `NameToNumber` storage is the one missing items.  Need to iterate over the `Portfolios` (number to name) map and do an insert for each missing name missing from the `NameToNumber` storage.  Also it would be nice to have a total count the number of missing names.